api/deployment/get.go

```diff
@@ -85,15 +92,21 @@ func GetDeployment(c *gin.Context) {
 		return
 	}
 
-	// send API call to capture the deployment
-	d, err := scm.FromContext(c).GetDeployment(ctx, u, r, int64(number))
+	// send API call to database to capture the deployment
+	d, err := database.FromContext(c).GetDeployment(int64(number))
 	if err != nil {
-		retErr := fmt.Errorf("unable to get deployment %s: %w", entry, err)
+		// send API call to SCM to capture the deployment
+		dep, err = scm.FromContext(c).GetDeployment(ctx, u, r, int64(number))
+		if err != nil {
+			retErr := fmt.Errorf("unable to get deployment %s: %w", entry, err)
 
-		util.HandleError(c, http.StatusInternalServerError, retErr)
+			util.HandleError(c, http.StatusInternalServerError, retErr)
 
-		return
+			return
+		}
+	} else {
+		dep = d
 	}
 
```

Comment:
we should safety check here since nil is the original value
```suggestion
if dep == nil {
// return errors.New("deployment for xyz is nil")
```

Reply:
done!